### ISSUE TITLE
Upcoming release management

### ISSUE THREAD

**Issue by dagwieers** [emilsvennesson/script.module.inputstreamhelper#85]
So, how do we want to proceed now?

I envisioned that we could work first on making the existing functionality better and fixing bugs. And adding unit and integration tests. So that the next release (v04.0) is a good foundation to think about new functionality.

But we are almost at the point where we have done everything we set out to do. And the biggest need of users (especially when things are not working as expected) is to control the exact Widevine version, (force) reinstall Widevine, etc.

And IMO there is no dire need to make a new release now, especially if it doesn't bring these additional feature to the table. So maybe we should start the discussion about how we expect addon developers and users to interact with this *plugin*.

- A **Configuration** settings category to:
  - ~Enable/disable inputstreamhelper~ #34 
  - Configure the frequency of Widevine checking for new releases
  - Maybe a way to influence the HWID to be used (auto, or manually configured) (ARM)
  - Configure the temporary location for download and extraction (ARM) #21 
- A **Widevine** settings category to:
  - ~Install, reinstall, upgrade, downgrade, remove Widevine~ #52 #91
- An **Info** settings category with information on the currently installed Widevine/ChromeOS
- End-user docs in Wiki on how to troubleshoot and fix Widevine issues
- ~An API for developers to integrate with~ #91
- Add-on developer docs in Wiki on how add-ons should refer to our settings-page

**Comment by dagwieers** [comment]
Other features we could add:
- A non-interactive option, when set, inputstreamhelper will just upgrade without asking questions
- An option to stick to a specific version
- An option that would only upgrade when Widevine fails (revoked or incompatible)

**Comment by dagwieers** [comment]
cc @emilsvennesson @mediaminister @horstl @asciidisco @vlmaksime @cache-sk @matthuisman @basrieter

**Comment by horstle** [comment]
I think now is a good time to create a new release (maybe just 0.3.6 since there's not much change visible to user). After that we could start implementing the stuff from 0.5.0 milestone and also improve communication to the user (the only thing left from the 0.4.0 milestone).
Anyway I would really like to see my PRs merged soon, as it's a bit annoying constantly replacing the IDs of the strings during a rebase, because another PR added a string with the same ID...

**Comment by dagwieers** [comment]
@horstle Before we can do a release we need to implement #76 first and then translate as much as possible. (Include past translators?) And include sufficient testing. We probably need a test-plan to ensure we have everything covered and tested before release.

**Comment by mediaminister** [comment]
In my opinion there is no urgent need for a release.
I think we should focus on performance and automation before releasing a new version.
In the end an average user just want to watch his favorite tv show and there should as little as possible interaction between this script.module and the user.
We should only interact with users if something fails and offer a manual alternative if something is broken.

**Comment by dagwieers** [comment]
> there should as little as possible interaction between this script.module and the user.

I agree with that. If anything our efforts should be focussed on seamless execution. My only concern is an important fix of which I don't know how many people have been impacted by that bug #79.

**Comment by mediaminister** [comment]
Okay, I didn't take into account https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/79, but a crash is definitely a reason to release.
So, we should test what we have now and maybe release next week?

**Comment by dagwieers** [comment]
@mediaminister Ok, let's do this. I do think we ought to call this **v0.4.0** because we made important changes in how it works wrt. ARM processing and how it appears to the user. Especially since we still need to redo those windows #76 (which IMO is essential now before retranslating everything). This is not just a bug-fix release.

So still on our list for v0.4.0:
- Communicate better to the user #76 
- Redesign settings now that they exist (move Disable to Troubleshooting section)
- Create a test plan #90 
- Retranslate (provide a small window for translations, subsequent translations go into v0.4.1)
- Appeal to create a test-group that covers our test plan

cc @emilsvennesson

**Comment by dagwieers** [comment]
> Anyway I would really like to see my PRs merged soon, as it's a bit annoying constantly replacing the IDs of the strings during a rebase, because another PR added a string with the same ID...

I feel sorry for asking you to resolve conflicts and then causing new conflicts :-/ Translations will always cause merge conflicts if you have more than one open PR and things get merged. The obvious solution is that we need to get features accepted faster, or request to keep English strings (and implement the translations ourselves after/during merge). Soon there is an end to the suffering ;-)

**Comment by dagwieers** [comment]
I moved the ideas from this thread to the Wiki at: https://github.com/emilsvennesson/script.module.inputstreamhelper/wiki/Ideas

**Comment by dagwieers** [comment]
I am moving the release information to #104, so we are almost ready for liftoff.
Please help with testing the current master branch and translations.
